File: website/adminscripts/changePassword.js
const path = require("path");
const sqlite3 = require('sqlite3').verbose(); //https://www.npmjs.com/package/sqlite3
const sha1 = require('sha1'); //https://www.npmjs.com/package/sha1
const siteconfig = require("../siteconfig").load();

if (process.argv.length != 3) {
    console.log("This tool changes the password for user with a given email")
    console.log(`Usage: node changePassword.js email`);
    process.exit(1);
}

var dbFile = path.join(siteconfig.dataDir, siteconfig.dbFile);
var db = new sqlite3.Database(dbFile, function(err){
  if (err) {
      console.log("ERR!", err)
      return
  }
  console.log('Connected to ' + dbFile + ' database.')
});

var email = process.argv[2];
var password = Math.random().toString(36).slice(-10);
var passwordHash = sha1(password);

db.run(
  "update users set passwordHash = $passwordHash where email = $email",
  { $email: email, $passwordHash: passwordHash },
  function(err) {
    if(err) {
      console.log("Could not update password for user with email: ", email);
    }
    else {
      console.log("User with email ", email, " now has a new password: ", password);
    }
  }
);

db.close();

File: website/adminscripts/changePassword.py
#!/usr/bin/env python3

import json
import sys
import os.path
import sqlite3
import hashlib
import random
import string

if len(sys.argv) != 2:
    print("This tool changes the password for user with a given email")
    print("Usage: ./adminscripts/changePassword.py email")
    sys.exit()

siteconfig = json.load(open(os.environ.get("LEXONOMY_SITECONFIG",
                                           "siteconfig.json"), encoding="utf-8"))
path = os.path.join(siteconfig["dataDir"], 'lexonomy.sqlite')
conn = sqlite3.connect(path)
print("Connected to database: %s" % path)

email = sys.argv[1];
password = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(8))
passhash = hashlib.sha1(password.encode("utf-8")).hexdigest();
try:
    conn.execute("update users set passwordHash=? where email=?", (passhash, email))
    conn.commit()
    print("User %s now has a new password: %s" % (email, password))
except sqlite3.Error as e:
    print("Could not update password for user %s. Database error: %s" % (email, e))

conn.close() 

